src/utils/self_test/SConscript

```diff
@@ -13,15 +13,16 @@ SELF_TEST = 'self_test.c'
 def scons():
     """scons function"""
 
-    Import('env', 'prereqs')
+    Import('env', 'prereqs', 'cart_lib', 'gurt_lib', 'cart_utils_objs')
 
     tenv = env.Clone(LIBS=[])
 
     tenv.AppendUnique(LIBS=['cart', 'gurt', 'pthread', 'm'])
     prereqs.require(tenv, 'mercury')
     tenv.AppendUnique(FLAGS='-pthread')
 
-    self_test = daos_build.program(tenv, SELF_TEST)
+    self_test = daos_build.program(tenv, [SELF_TEST, cart_utils_objs])
+    tenv.Requires(self_test, [cart_lib, gurt_lib])
```

Comment:
another requires line.   I would test fix all of the CPPPATH and LIBPATH issues, make a commit, then remove these and see if build still works...  If not, defer cleanup or Requires to another patch.

Reply:
CI seems to [build/run okay](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-6115/4) [without the Requires](https://github.com/daos-stack/daos/pull/6115/commits/95f424939288f9a595a9ffc5a25ab39b5ef1b0e5).